payments/schema/queries.py

```diff
@@ -61,3 +65,27 @@ def resolve_additional_products(self, info, **kwargs):
                 )
 
         return AdditionalProduct.objects.all()
+
+    def resolve_additional_product_services(self, info, **kwargs):
+        services = []
+        service_list = []
+        product_type = kwargs.get("product_type")
+        if product_type:
+            product_type = AdditionalProductType(product_type)
+            if product_type == AdditionalProductType.FIXED_SERVICE:
+                service_list = ServiceType.FIXED_SERVICES()
+            elif product_type == AdditionalProductType.OPTIONAL_SERVICE:
+                service_list = ServiceType.OPTIONAL_SERVICES()
+        else:
+            service_list = list(ServiceType)
+
+        for service in service_list:
+            services.append(
+                AdditionalProductServiceNode(
+                    service=service,
+                    product_type=AdditionalProductType.FIXED_SERVICE
+                    if service.is_fixed_service()
+                    else AdditionalProductType.OPTIONAL_SERVICE,
```

Comment:
maybe a more generic way of figuring out `product_type` would be to have a resolver on the node?
```python
class AdditionalProductServiceNode(graphene.ObjectType):
    service = ServiceTypeEnum(required=True)
    product_type = AdditionalProductTypeEnum(required=True)

    class Meta:
        interfaces = (graphene.relay.Node,)
        connection_class = CountConnection
    
    def resolve_product_type(self, info):
        return AdditionalProductType.FIXED_SERVICE if self.service.is_fixed_service() else AdditionalProductType.OPTIONAL_SERVICE
```

this wayyou can only pass the service whenever you need to init the node

Reply:
also you could skip initing `services` list and just do:
```python
def resolve_additional_product_services(self, info, **kwargs):
    service_list = list(ServiceType)
    product_type = kwargs.get("product_type")
    if product_type:
        if product_type == AdditionalProductType.FIXED_SERVICE:
            service_list = ServiceType.FIXED_SERVICES()
        elif product_type == AdditionalProductType.OPTIONAL_SERVICE:
            service_list = ServiceType.OPTIONAL_SERVICES()
    return [AdditionalProductServiceNode(service=service) for service in service_list]
```

Reply:
apart from these two nitpicks, a solid implementation of the enum query 👍 👍 👍